pkg/agent/multicast/mcast_controller.go

```diff
@@ -473,6 +483,46 @@ func (c *Controller) addOrUpdateGroupEvent(e *mcastGroupEvent) {
 	}
 }
 
+func (c *Controller) memberChanged(e interface{}) {
+	podEvent := e.(types.PodUpdate)
+	namespace, name := podEvent.PodNamespace, podEvent.PodName
+
+	klog.V(2).InfoS("Pod is updated", "IsAdd", podEvent.IsAdd, "namespace", namespace, "name", name)
+	event := &mcastGroupEvent{
+		group: types.McastAllHosts,
+	}
+	c.groupEventCh <- event
+	c.removeLocalInterface(podEvent)
+}
+
+func (c *Controller) initQueryGroup() error {
+	err := c.updateQueryGroup()
+	if err != nil {
+		return err
+	}
+	if err = c.ofClient.InstallMulticastFlows(types.McastAllHosts, c.queryGroupId); err != nil {
+		klog.ErrorS(err, "Failed to install multicast flows", "group", types.McastAllHosts)
+		return err
+	}
+	return nil
+}
+
+// updateQueryGroup gets all containers' interfaces, and add all ofports into IGMP query group.
+func (c *Controller) updateQueryGroup() error {
+	ifaces := c.ifaceStore.GetInterfacesByType(interfacestore.ContainerInterface)
+	memberPorts := make([]uint32, 0, len(ifaces))
+	for _, iface := range ifaces {
+		memberPorts = append(memberPorts, uint32(iface.OFPort))
+	}
+	// Install OpenFlow group for a new multicast group which has local Pod receivers joined.
+	if err := c.ofClient.InstallMulticastGroup(c.queryGroupId, true, memberPorts); err != nil {
```

Comment:
I think it's more readable to append `HostGatewayOFPort` to memberPorts on caller side, `isIGMPQueryGroup` makes the method too specific to caller side logic and invaded.

Reply:
done